Vertex AI provider returns 404 for region: global

**Describe the bug**

The Vertex AI provider generates an incorrect API host when `region: global` is specified, resulting in 404 errors.

The provider attempts to construct the hostname using the pattern `{region}-aiplatform.googleapis.com`, which works for regional endpoints but fails for the global region. The hostname `global-aiplatform.googleapis.com` does not exist. For the global region, the correct endpoint is `aiplatform.googleapis.com` (without any region prefix).

This issue particularly affects users who want to access preview models like Gemini 3 Pro Preview, which are often released to the global endpoint before regional availability.

**To Reproduce**

1. Create a promptfoo configuration file with `region: global`:

```yaml
providers:
- id: vertex:gemini-3-pro-preview
  config:
    region: global
    projectId: your-project-id

prompts:
- "Hello"

tests:
- vars: {}
```

2. Run the evaluation:

```bash
promptfoo eval -c config.yaml
```

3. Observe the 404 error

```
[ERROR] API call error: Error: <!DOCTYPE html>
<html lang=en>
  <title>Error 404 (Not Found)!!1</title>
```

The provider constructs an incorrect URL:
```
https://global-aiplatform.googleapis.com/v1/projects/{projectId}/locations/global/publishers/google/models/{model}:streamGenerateContent
```

The hostname `global-aiplatform.googleapis.com` does not exist.


**Expected behavior**

The provider should successfully call the Vertex AI API using the correct global endpoint (`aiplatform.googleapis.com`) and return a response from the model.

**Screenshots**

N/A


**System information:**

Promptfoo version: 0.119.10 &  also present in current main branch

**Additional context**

**Workaround**

Explicitly set the `apiHost` in the provider config:

```yaml
providers:
- id: vertex:gemini-3-pro-preview
  config:
    region: global
    apiHost: aiplatform.googleapis.com
    projectId: your-project-id
```

This workaround resolves the issue for generative models. However, the provider should automatically handle the global region case to improve user experience and avoid confusion.

The hostname pattern differs between regional and global endpoints:
Regional: `{region}-aiplatform.googleapis.com` (e.g., `us-central1-aiplatform.googleapis.com`)
Global: `aiplatform.googleapis.com` (no region prefix)

--- a/src/providers/google/vertex.ts
+++ b/src/providers/google/vertex.ts
@@ -45,6 +45,19 @@ import type {
 // Type for Google API errors - using 'any' to avoid gaxios dependency
 type GaxiosError = any;
 
+function getVertexApiHost(
+  region: string,
+  configApiHost?: string,
+  envOverrides?: EnvOverrides,
+): string {
+  return (
+    configApiHost ||
+    envOverrides?.VERTEX_API_HOST ||
+    getEnvString('VERTEX_API_HOST') ||
+    (region === 'global' ? 'aiplatform.googleapis.com' : `${region}-aiplatform.googleapis.com`)
+  );
+}
+
 class VertexGenericProvider implements ApiProvider {
   modelName: string;
 
@@ -70,13 +83,8 @@ class VertexGenericProvider implements ApiProvider {
     return `[Google Vertex Provider ${this.modelName}]`;
   }
 
-  getApiHost(): string | undefined {
-    return (
-      this.config.apiHost ||
-      this.env?.VERTEX_API_HOST ||
-      getEnvString('VERTEX_API_HOST') ||
-      `${this.getRegion()}-aiplatform.googleapis.com`
-    );
+  getApiHost(): string {
+    return getVertexApiHost(this.getRegion(), this.config.apiHost, this.env);
   }
 
   async getProjectId(): Promise<string> {
@@ -1009,6 +1017,10 @@ export class VertexEmbeddingProvider implements ApiEmbeddingProvider {
     return this.config.apiVersion || 'v1';
   }
 
+  getApiHost(): string {
+    return getVertexApiHost(this.getRegion(), this.config.apiHost, this.env);
+  }
+
   async getProjectId(): Promise<string> {
     return await resolveProjectId(this.config, this.env);
   }
@@ -1030,7 +1042,7 @@ export class VertexEmbeddingProvider implements ApiEmbeddingProvider {
     try {
       const client = await this.getClientWithCredentials();
       const projectId = await this.getProjectId();
-      const url = `https://${this.getRegion()}-aiplatform.googleapis.com/${this.getApiVersion()}/projects/${projectId}/locations/${this.getRegion()}/publishers/google/models/${
+      const url = `https://${this.getApiHost()}/${this.getApiVersion()}/projects/${projectId}/locations/${this.getRegion()}/publishers/google/models/${
         this.modelName
       }:predict`;
       const res = await client.request<any>({

--- a/test/providers/google/vertex.test.ts
+++ b/test/providers/google/vertex.test.ts
@@ -2259,4 +2259,39 @@ describe('VertexChatProvider.callClaudeApi parameter naming', () => {
       );
     });
   });
+
+  describe('getApiHost', () => {
+    it('should return global endpoint without region prefix for region: global', () => {
+      const provider = new VertexChatProvider('gemini-pro', { config: { region: 'global' } });
+      expect(provider.getApiHost()).toBe('aiplatform.googleapis.com');
+    });
+
+    it('should return regional endpoint for non-global regions', () => {
+      const provider = new VertexChatProvider('gemini-pro', { config: { region: 'us-central1' } });
+      expect(provider.getApiHost()).toBe('us-central1-aiplatform.googleapis.com');
+    });
+
+    it('should use custom apiHost over default', () => {
+      const provider = new VertexChatProvider('gemini-pro', {
+        config: { region: 'global', apiHost: 'custom.example.com' },
+      });
+      expect(provider.getApiHost()).toBe('custom.example.com');
+    });
+
+    it('should use VERTEX_API_HOST from env override', () => {
+      const provider = new VertexChatProvider('gemini-pro', {
+        config: { region: 'global' },
+        env: { VERTEX_API_HOST: 'env.example.com' },
+      });
+      expect(provider.getApiHost()).toBe('env.example.com');
+    });
+
+    it('should prioritize configApiHost over env override', () => {
+      const provider = new VertexChatProvider('gemini-pro', {
+        config: { region: 'global', apiHost: 'config.example.com' },
+        env: { VERTEX_API_HOST: 'env.example.com' },
+      });
+      expect(provider.getApiHost()).toBe('config.example.com');
+    });
+  });
 });

